src/py/flwr/common/address_test.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2020 Adap GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower IP address utils."""
+
+
+from .address import parse_address
+
+
+def test_ipv4_correct() -> None:
+    """Test if a correct IPv4 address is correctly parsed."""
+
+    # Prepare
+    addresses = [
+        ("127.0.0.1:8080", ("127.0.0.1", 8080, False)),
+        ("0.0.0.0:12", ("0.0.0.0", 12, False)),
+    ]
+
+    for address, expected in addresses:
+        # Execute
+        actual = parse_address(address)
+
+        # Assert
+        assert actual == expected
+
+
+def test_ipv4_incorrect() -> None:
+    """Test if an incorrect IPv4 address returns None."""
+
+    # Prepare
+    addresses = [
+        "127.0.0.1::8080",
+        "0.0.0.0.0:12",
+        "1112.0.0.0:50",
+        "127.0.0.1",
+        "42.1.1.0:9988898",
+    ]
+
+    for address in addresses:
+        # Execute
+        actual = parse_address(address)
+
+        # Assert
+        assert actual is None
+
+
+def test_ipv6_correct() -> None:
+    """Test if a correct IPv6 address is correctly parsed."""
+
+    # Prepare
+    addresses = [
```

Comment:
@tanertopal, same question as above: are there any other cases we need to consider?